examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,136 @@
+#include "static-supported-modes-manager.h"
+#include "ESP32Config.h"
+#include <app/util/debug-printing.h>
+#include <app/util/ember-print.h>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer::Internal;
+using namespace chip::app::Clusters::ModeSelect;
+using chip::Protocols::InteractionModel::Status;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+using SemanticTag          = Structs::SemanticTagStruct::Type;
+template <typename T>
+using List                                                                                = app::DataModel::List<T>;
+
+StaticSupportedModesManager::ModeLabel* StaticSupportedModesManager::modeLabelList = nullptr;
+StaticSupportedModesManager::ModeOptionStructType* StaticSupportedModesManager::modeOptionStruct = nullptr;
+StaticSupportedModesManager::SemanticTag* StaticSupportedModesManager::semanticTags = nullptr;
+
+const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
+
+SupportedModesManager::ModeOptionsProvider StaticSupportedModesManager::getModeOptionsProvider(EndpointId endpointId) const
+{
+    char keyBuf[ESP32Config::kMaxConfigKeyNameLength];
+    uint32_t supportedModeCount = 0;
+
+    VerifyOrReturnValue(ESP32Config::KeyAllocator::SupportedModesCount(keyBuf, sizeof(keyBuf), endpointId) == CHIP_NO_ERROR,
+                        ModeOptionsProvider(nullptr, nullptr));
+    ESP32Config::Key countKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+    VerifyOrReturnValue(ESP32Config::ReadConfigValue(countKey, supportedModeCount) == CHIP_NO_ERROR,
+                        ModeOptionsProvider(nullptr, nullptr));
+
+	modeLabelList = new ModeLabel[supportedModeCount];
```

Comment:
OK, so every call to `getModeOptionsProvider()` allocates a new list here.

But there is only one delete, in ~SupportedModesManager.  So if `getModeOptionsProvider()` is called twice you leak.

In particular, if `getModeOptionByMode` is called twice you leak.